internal/install/recipe_installer.go

```diff
@@ -169,20 +170,23 @@ Welcome to New Relic. Let's set up full stack observability for your environment
 		"RecipeNamesProvided": i.RecipeNamesProvided(),
 	}).Debug("context summary")
 
+	if i.RecipeNamesProvided() {
+		i.status.SetTargetedInstall()
+	}
+
 	ctx, cancel := context.WithCancel(utils.SignalCtx)
 	defer cancel()
 
 	errChan := make(chan error)
 	var err error
 
 	err = i.connectToPlatform()
-
 	if err != nil {
-		return err
-	}
+		if _, ok := err.(*nrErrors.PaymentRequiredError); ok {
+			i.status.InstallComplete(err)
```

Comment:
It doesn't seem right to have an install complete event without a started event before that. I understand why you've added the complete here though (to get the metric on how many users are having this paywall error), but perhaps the start event should go first.
What happen if there are other NR errors, such as network issues? Perhaps in those cases we shouldn't attempt to write status events?

Reply:
Good point. I think I had the `InstallStarted` call before this at one point. I'll take a look again and see what I can come up with.

Reply:
Yep, I had moved this up at one point. I just moved it to where I had it before (just before call `status.SetTargetedInstall()`)